### PR TITLE
WIP: Add dplyr join content

### PR DESCRIPTION
**Pull request by JulieHawkins**
creating joins chapter for course

### CODE REVIEW

**Comment by JulieHawkins** [comment]
joins chapter content is now finished - please code review and add to master if ok

File: xx-joins.Rmd [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+objectives:
+- "To understand the concepts of relational data and be able to change one table based on another with dplyr's `xxx_join()` functions."
+keypoints:
+- "Use the `dplyr` package to join relational data."
+- "Use `left_join()`, `full_join()`, `right_join()` or `inner_join()` to add data from one table to another."
+- "Use `semi_join()` or `anti_join()` to filter a table based on data in another table."
+source: Rmd
+---
+
+# Relational Data
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:16]: `# Joining Related Data`

File: xx-joins.Rmd [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+```{r, include=FALSE}
+source("bin/chunk-options.R")
+knitr_fig_path("xx-")
+```
+
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:22]: ## Relational Data

File: xx-joins.Rmd (hunk 1 above)
Comment (BonnieJRobert) [inline xx-joins.Rmd:16]:
```suggestion
# Joining Related Data
```

File: xx-joins.Rmd (hunk 2 above)
Comment (BonnieJRobert) [inline xx-joins.Rmd:22]:
```suggestion
## Introducing Relational Data
```

File: xx-joins.Rmd [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+In real life, we often have two or more related datasets that have a key 
+linking the data.
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:24]:
```suggestion
In real life, we often have two or more related datasets that share a set of common columns.   We would like to combine these two datasets based on their common columns, also called a **key** or **key variables**.
```

File: xx-joins.Rmd [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+
+We may want to change a dataset by adding information from a second dataset.
+For example, one dataset may contain employee information such as employee ID, 
+name, job position, hire date, etc., whereas a second dataset may contain the
+current and historical wages for each job position. A relevant join -- adding a
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:29]:
```suggestion
For example, say we have a dataset containing employee information such as employee ID, 
name, job position, hire date, etc., and a second dataset containing the
current and historical wages for each job position. A relevant join -- adding a 
```

File: xx-joins.Rmd [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+For example, one dataset may contain employee information such as employee ID, 
+name, job position, hire date, etc., whereas a second dataset may contain the
+current and historical wages for each job position. A relevant join -- adding a 
+column of data for each row -- may be merging the current wage for each employee 
+into the employee dataset.
+
+We may also want to choose only a portion of a dataset. For example, one dataset
+may include all students who ever attended a school along with their student ID, 
+years attended, grades for each class, graduation date (if applicable), and so on,
+whereas a second dataset may be simply a list of current students. A relevant join
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:36]:
```suggestion
We may also want to choose only a portion of a dataset. For example, say we have a dataset including all students who ever attended a school along with their student ID, 
years attended, grades for each class, graduation date (if applicable), and so on,
and a second dataset which is simply a list of current students. A relevant join
```

File: xx-joins.Rmd [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+
+## Types of joins with dplyr
+
+The `dplyr` package has several functions that allow us to make these joins. 
+Assume we have tables _x_ and _y_.
+
+There are four types of **mutating joins**, which add new variables to one 
+table from matching rows in another:
+
+* **inner joins** include all rows in x AND y
```
Comment (BonnieJRobert) [inline xx-joins.Rmd:55]: I'd be tempted to expand the vignette descriptions to include the matching rows.  Something like:
**left_joins** include all rows in x *plus* any row in y with a match to a row in x

**Review by BonnieJRobert — COMMENTED** [review]
Nice work, Julie!  Super clean code and easy to follow.  I've added a few suggestions to help clarify definitions for ppl who are reading the online materials, only.

**Pull request merged**